src/utils/utils.ts

```diff
@@ -204,3 +209,100 @@ export async function retry<T = any>(
     process.exit()
   }
 }
+
+export type SaleConfig = {
+  publicSalePrice: ethers.BigNumber
+  maxSalePurchasePerAddress: number
+  publicSaleStart: number
+  publicSaleEnd: number
+  presaleStart: number
+  presaleEnd: number
+  presaleMerkleRoot: string
+}
+
+type SalesConfigTuple = [
+  string, // erc721TransferHelper
+  string, // marketFilterAddress
+  string, // initialOwner
+  string, // fundsRecipient
+  number, // number of editions
+  number, // percentage of royalties in bps
+  SaleConfig, // sales config object
+  string, // metadataRenderer
+  string, // metadataRendererInit
+]
+
+type HolographERC721ConfigTuple = [
+  string, // contractName
+  string, // contractSymbol
+  number, // contractBps
+  string, // eventConfig
+  boolean, // skipInit
+  string, // initializer
+]
+
+export function generateSalesConfigTuple(
```

Comment:
I love all this new stuff. Rather than have in utils. Can we make a new file for these kinds of things? Maybe `byteGenerators`, `tupleGenerator`, `byteConstruction`, `byteCodeGenerator` ? Open to other ideas, but i think specific file for these kinds of things is better than the generic 'utils' file.